### ISSUE TITLE
cylc broadcast: 6.4.0+ client can crash a 6.3.1 suite

### ISSUE THREAD

**Issue by benfitzpatrick** [cylc/cylc#1431]
When a (cylc 6 syntax) suite is running at cylc 6.3.1, and a cylc 6.4.0+ 'cylc broadcast' (that is otherwise correct) is run that looks like this:
```
cylc broadcast -s [environment]FOO=true -p 20130101T0000Z SOME_SUITE
```
then it brings down the daemon with:
```
Traceback (most recent call last):
  File "/opt/cylc/lib/cylc/run.py", line 75, in main
    server.run()
  File "/opt/cylc/lib/cylc/scheduler.py", line 910, in run
    if flags.debug:
  File "/opt/cylc/lib/cylc/broadcast.py", line 129, in expire
    
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 116, in __cmp__
    return self._iso_point_cmp(self.value, other.value)
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 85, in _wrapper
    results = function(*args)
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 154, in _iso_point_cmp
    point = point_parse(point_string)
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 695, in point_parse
    return _point_parse(point_string).copy()
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 85, in _wrapper
    results = function(*args)
  File "/opt/cylc/lib/cylc/cycling/iso8601.py", line 723, in _point_parse
    "non-ISO-8601-compatible cycle point: %s" % point_string
  File "/opt/cylc/lib/cylc/syntax_flags.py", line 68, in set_syntax_version
    raise exc_class(*exc_args, **exc_kwargs)
SyntaxVersionError: Conflicting syntax: post-cylc-6 syntax (cycle point: [scheduling]initial cycle point = 20130101T00) vs pre-cylc-6 syntax (non-ISO-8601-compatible cycle point: root)
ERROR CAUGHT: cleaning up before exit
2015-04-20T13:38:54Z WARNING - some active tasks will be orphaned
THE ERROR WAS:
Conflicting syntax: post-cylc-6 syntax (cycle point: [scheduling]initial cycle point = 20130101T00) vs pre-cylc-6 syntax (non-ISO-8601-compatible cycle point: root)
use --debug to turn on exception tracebacks)
```

**Comment by benfitzpatrick** [comment]
The relevant change is https://github.com/cylc/cylc/pull/1388.

**Comment by matthewrmshin** [comment]
@benfitzpatrick I am removing the bug label because this is a version incompatibility issue instead of a bug in a cylc version.

I am not sure what is the best action here given that it is the server that dies. The easiest fix is to perform a version check between client and server, and fails the client on incompatible versions. (We can perhaps make this `global.rc` configurable to demote the client failure to a warning.) Otherwise, we'll need an awkward version compat logic for the broadcast client, or simply reverse the swap of the argument list.

For the future, after #1395 is merged, we can have the client send its version to the server, and the server can then reject any commands from incompatible clients.

**Comment by hjoliver** [comment]
Actually #1388 is more generally not backward compatible with older suite daemons in that the broadcast methods now return more information than they used to but the broadcast command does not handle the old form.  We normally do handle this sort of scenario properly (although it arises more frequently in the GUI code than the CLI).  @matthewrmshin - can you possibly fix this before the next release?

**Comment by matthewrmshin** [comment]
Sorry, it has been a while since I looked through the issue list, and found this one. Do we still want any action for this one?

**Comment by hjoliver** [comment]
I'm happy close this as won't fix, it is unlikely to affect anyone now.

**Comment by benfitzpatrick** [comment]
Yep, won't fix.
